metadata/reader.go

```diff
@@ -317,12 +330,12 @@ func (r *reader) getOrCreateDir(nodes *bolt.Bucket, md map[uint32]*metadataEntry
 		if err := writeAttr(b, attr); err != nil {
 			return 0, nil, err
 		}
-		if d != "" {
-			pid, pb, err := r.getOrCreateDir(nodes, md, parentDir(d), rootID)
+		if dir != "" {
```

Comment:
> Before a root file would have its parent be set to the empty string, now it will be "/".
 
The nodes are structured like an m-ary tree, where in each node only contains pointers to its children referenced by `id`. We hit a base case when we find a parent that exists in the tree. Given a path `a/b/c`, where in only the root exists:

```
Initial
path = `a/b/c`
Does `c` exist? No, generate node ID for `c` and recurse

recurse(1) returns set `c` to be a child of id of `b` and finally return the node id for `c`

Recurse(2)
path = `a/b`
Does `b` exist? No, generate node ID for `b` and recurse

recurse(3) returns set `b` to be a child of id of `a`

Recurse(3)
path = `a`
Does `a` exist? No, generate node ID for `a` and recurse

recurse(4) returns set  `a` to be a child of root id
                                                                                                      
Recurse(4)                                                  
path = ``                                                     
Does `` exist? Yes return root id 

```


> So dir will never be empty here.

In fact, I think this check here is useless, since we hit a definite base case when the path is `""`, meaning the root always exists.